controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -93,46 +89,54 @@ public void refresh() {
     }
 
     @Override
-    CompletableFuture<Void> checkStreamExists(Create create) {
-        CompletableFuture<Void> result = new CompletableFuture<>();
+    CompletableFuture<CreateStreamResponse> checkStreamExists(StreamConfiguration configuration, long timestamp) {
```

Comment:
its not checking stream creation state.. its checking if stream exists or not.. 
if it does, this creates a response where it puts in "original" stream's metadata so that subsequent steps can use the original data.